webapp/graphite/render/datalib.py

```diff
@@ -111,11 +114,60 @@ def fetchData(requestContext, pathExpr):
   startTime = int( epoch( requestContext['startTime'] ) )
   endTime   = int( epoch( requestContext['endTime'] ) )
 
+  def _fetchDataWrapper(node, startTime, endTime, result_queue):
+    result = node.fetch(startTime, endTime)
+    result_queue.put((node, result))
+
   def _fetchData(pathExpr,startTime, endTime, requestContext, seriesList):
+    result_queue = Queue.Queue()
```

Comment:
I'm a little confused about the pre-existing threading model and these changes. Are these fetches queued up / threaded in some other place? Or was this a missed opportunity that you're now fixing?